test/protected_resource_test.rb

```diff
@@ -0,0 +1,90 @@
+require 'test_helper'
+
+class TestResource < Minitest::Test
+  SomeError = Class.new(StandardError)
+
+  def setup
+    Semian.destroy(:testing)
+  rescue
+    nil
+  end
+
+  def teardown
+    destroy_resources
+  end
+
+  def test_acquire_without_bulkhead
+    Semian.register(
+      :testing,
+      exceptions: [SomeError],
+      error_threshold: 2,
+      error_timeout: 5,
+      success_threshold: 1,
+      bulkhead: false,
+    )
+    @resource = Semian[:testing]
+    assert_instance_of Semian::CircuitBreaker, @resource.circuit_breaker
+    assert_nil @resource.bulkhead
+  end
+
+  def test_acquire_without_circuit_breaker
+    Semian.register(
+      :testing,
+      tickets: 2,
+      circuit_breaker: false,
+    )
+    acquired = false
+
+    @resource = Semian[:testing]
+    @resource.acquire do
```

Comment:
This doesn't actually test that the circuit breaker isn't used from an integration point of view. Perhaps you should try to do 10 failures and make sure you can still acquire it, or something like this?